iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -53,15 +54,15 @@ private enum StateFlags
         private const char SegmentSeparatorChar = '/';
         private const char SingleSegmentWildcardChar = '+';
         private const char MultiSegmentWildcardChar = '#';
-        private static readonly char[] s_wildcardChars = { MultiSegmentWildcardChar, SingleSegmentWildcardChar };
         private const string IotHubTrueString = "true";
         private const string SegmentSeparator = "/";
         private const int MaxPayloadSize = 0x3ffff;
         private const int MaxTopicNameLength = 0xffff;
 
+        private static readonly char[] s_wildcardChars = { MultiSegmentWildcardChar, SingleSegmentWildcardChar };
         private static readonly Action<object> s_pingServerCallback = PingServer;
         private static readonly Action<object> s_checkConnAckTimeoutCallback = ShutdownIfNotReady;
-        private static readonly Func<IChannelHandlerContext, Exception, bool> s_shutdownOnWriteErrorHandler = (ctx, ex) => { ShutdownOnError(ctx, ex); return false; };
+        private static readonly Func<IChannelHandlerContext, Exception, Task<bool>> s_shutdownOnWriteErrorHandler = async (ctx, ex) => { await ShutdownOnErrorAsync(ctx, ex).ConfigureAwait(true); return false; };
```

Comment:
isn't using ConfigureAwait(true) here not needed? As in, removing it is like having the same effect.

Reply:
aaah..it could be because you're trying to suppress a warning

Reply:
It is a function that will be called in a method below, so it needs to maintain the same async strategy used elsewhere in the file.

Before, it was unawaited.